lib/grp.gi

```diff
@@ -4401,6 +4401,13 @@ InstallMethod( GroupWithGenerators,
 function( gens )
 local G,typ;
 
+  if IsGroup(gens) then
+    Info( InfoPerformance, 1,
+      "Calling `GroupWithGenerators' on a group usually is very inefficient.");
+    Info( InfoPerformance, 1,
+      "Use the list of generators of the group instead.");
```

Comment:
But isn't the actual problem that using a large number of generators is usually inefficient? Then, shouldn't the check instead be on the *size* of `gens`? And then, shouldn't it apply equally to lists as to domains or any other kind of collection?

Reply:
But how then would you know when the number is large? I assume that the number of generators is usually smaller than the number of elements in the group (unless it is already given using a very inefficient list of generators).

Reply:
Sure, any number we'd pick as "large" would be arbitrary. Just like it is arbitrary to allow `Group(AsList(G))` as generators, but not `Group(G)`. So my personal preference would be to do neither, i.e., exactly as it is right now...

Reply:
Can't think of any scenario when calling `Group(Group(...))` is more efficient than `Group(GeneratorsOfGroup(...))` - that's what I am suggesting to catch. Not other inefficiencies, as that would be more ambiguous...